benches/storage_perf.rs

```diff
@@ -0,0 +1,73 @@
+use criterion::{Bencher, Criterion};
+use specs::prelude::*;
+
+#[derive(Clone, Debug)]
+pub struct GridVelocity {
+    pub dx: i16,
+    pub dy: i16,
+}
+impl GridVelocity {
+    pub fn new() -> Self {
+        GridVelocity { dx: 0, dy: 0 }
+    }
+}
+impl Component for GridVelocity {
+    type Storage = VecStorage<Self>;
+}
+
+fn bench_vec(b: &mut Bencher) {
+    let mut vels: Vec<GridVelocity> = (0..1000 * 1000).map(|_i| GridVelocity::new()).collect();
+    b.iter(|| {
+        vels.iter_mut().for_each(|vel| {
+            vel.dx += 1;
+        });
+    });
```

Comment:
You are basically comparing here iterating vector vs doing manually iteration with indices and using `get_unchecked` (the bench_vecstorage).
Have you tried to benchmark it directly on `Vec`?

Reply:
The difference might be because there are no `#[inline]` annotations on the `UnprotectedStorage` methods.

Reply:
I'm not sure get what you want me to benchmark ? if you're talking about indexing Vecs, it was quite similar to using iterators. Even using `as_mut_ptr()` didn't give the huge perf boost I expected.

Reply:
I was just wondering why `VecStorage` would differ at all from `Vec`.

Reply:
VecStorage and Vec have quite similar performance. Read/WriteStorage over VecStorage introduces the overhead